### PR TITLE
Feature/pwa

### LINKED ISSUE #225: PWA support
Look into making Watcharr a PWA.

### PR DESCRIPTION
**Pull request by ishaanparlikar**
Added PWA feature.

Closes #225

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,7 +30,8 @@
     "svelte-preprocess": "^5.1.1",
     "tslib": "^2.6.2",
     "typescript": "^5.3.2",
-    "vite": "^4.5.1"
+    "vite": "^4.5.1",
+    "workbox-window": "^7.0.0"
```
Comment (IRHM) [inline package.json]: @ishaanparlikar Looks like we don't use this dependency (workbox-window)? Are we able to remove it?

File: vite.config.ts [hunk 2]
```diff
@@ -1,11 +1,39 @@
 import { sveltekit } from "@sveltejs/kit/vite";
 import { defineConfig } from "vite";
 import { readFileSync } from "fs";
-
+import {SvelteKitPWA} from '@vite-pwa/sveltekit'
+/** @type {import('vite').UserConfig} */
```
Comment (IRHM) [inline vite.config.ts]: Can we remove this `@type` line?

**Review by IRHM — CHANGES_REQUESTED** [review]
Found a dependency I don't think is being used?

~Were you able to get the app to install in your browser? I think there should be an Install button in Firefox/chromium, but I don't see one anywhere. Not sure if I'm doing something wrong here.~

File: package.json (hunk 1 above)
Comment (ishaanparlikar) [reply package.json]: @IRHM  I was getting error before installing this dependancy. But I checked just now, I was able to build the project without the issue. I have removed it. Please have a look.
Thanks

File: vite.config.ts (hunk 2 above)
Comment (ishaanparlikar) [reply vite.config.ts]: @IRHM  Removed.

**Review by IRHM — APPROVED** [review]

**Comment by IRHM** [comment]
@ishaanparlikar Looks great, thanks again for getting this task done! I'll test and release this with the next release later tonight or tomorrow.

**Pull request merged**